sphinx/builders/linkcheck.py

```diff
@@ -109,9 +109,7 @@ def init(self) -> None:
                                for x in self.app.config.linkcheck_anchors_ignore]
         self.auth = [(re.compile(pattern), auth_info) for pattern, auth_info
                      in self.app.config.linkcheck_auth]
-        self.good = set()       # type: Set[str]
-        self.broken = {}        # type: Dict[str, str]
-        self.redirected = {}    # type: Dict[str, Tuple[str, int]]
```

Comment:
In my experience, the users of Sphinx sometimes hacks our modules. So it would be better to drop attribute carefully. So it would be better to deprecated them at first.
refs: https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/environment/__init__.py#L646-L660